internal/controller/nodeagent.go

```diff
@@ -470,6 +461,7 @@ func (r *DataProtectionApplicationReconciler) customizeNodeAgentDaemonset(ds *ap
 	if dpa.Spec.Configuration.NodeAgent.PodConfig != nil {
 		ds.Spec.Template.Spec.Tolerations = dpa.Spec.Configuration.NodeAgent.PodConfig.Tolerations
 		if len(dpa.Spec.Configuration.NodeAgent.PodConfig.NodeSelector) != 0 {
```

Comment:
do we need to remove this logic of adding NodeSelector to DaemonSet?

new configMap approach does not take care of this case?

Reply:
There are two problems we need to solve:
 1. We want to limit where the node-agent is scheduled
 2. We need to limit where the data mover pod is scheduled

The second must run on the node where node-agent is scheduled, that is why the second must be a subset of first.

Two of those are very much separate workflows and logic, because node-agent process uses CM and the node-agent pod uses affinity within DS.
We cannot have one without another, so now we do not want to remove it.